### ISSUE TITLE
Document has DockerRequirement under 'requirements' but use_container is false.

### ISSUE THREAD

**Issue by jeremiahsavage** [BD2KGenomics/toil#1200]
Using the most recent git [1] and a local cwltool [2], I get the following error when running a simple cwl [3]

Error:

```
(p2) [jeremiah@wireless-s1-so-150-52-240 ~]$ cwltoil ~/code/cocleaning-cwl/tools/md5sum.cwl ./ex.yml 
wireless-s1-so-150-52-240.uchicago.edu 2016-10-07 17:16:28,867 MainThread INFO toil.lib.bioio: Root logger is at level 'INFO', 'toil' logger at level 'INFO'.
wireless-s1-so-150-52-240.uchicago.edu 2016-10-07 17:16:29,195 MainThread INFO toil.lib.bioio: Root logger is at level 'INFO', 'toil' logger at level 'INFO'.
wireless-s1-so-150-52-240.uchicago.edu 2016-10-07 17:16:29,196 MainThread INFO toil.jobStores.fileJobStore: Path to job store directory is '/tmp/tmpNJ72E9'.
wireless-s1-so-150-52-240.uchicago.edu 2016-10-07 17:16:29,196 MainThread INFO toil.jobStores.abstractJobStore: The workflow ID is: '76b2b9e5-c80a-42da-a3a3-706e3905fff3'
Traceback (most recent call last):
  File "/home/jeremiah/.virtualenvs/p2/bin/cwltoil", line 11, in <module>
    load_entry_point('toil', 'console_scripts', 'cwltoil')()
  File "/home/jeremiah/code/toil/src/toil/cwl/cwltoil.py", line 653, in main
    builder = t._init_job(job, basedir=basedir)
  File "/home/jeremiah/.virtualenvs/p2/lib/python2.7/site-packages/cwltool/process.py", line 451, in _init_job
    raise WorkflowException("Document has DockerRequirement under 'requirements' but use_container is false.  DockerRequirement must be under 'hints' or use_container must be true.")
cwltool.errors.WorkflowException: Document has DockerRequirement under 'requirements' but use_container is false.  DockerRequirement must be under 'hints' or use_container must be true.

```

[1]
https://github.com/BD2KGenomics/toil/commit/196ba62d468963d57dc36be07e382c247551af62

[2]
https://github.com/jeremiahsavage/cwltool/releases/tag/1.0_gdc_e

[3]
md5sum.cwl:

```
#!/usr/bin/env cwl-runner

cwlVersion: v1.0

requirements:
  - class: DockerRequirement
    dockerPull: ubuntu:xenial-20160809
  - class: InlineJavascriptRequirement

class: CommandLineTool

inputs:
  - id: INPUT
    type: File
    inputBinding:
      position: 0

outputs:
  - id: OUTPUT
    type: File
    outputBinding:
      glob: $(inputs.INPUT.nameroot + ".md5")

stdout: $(inputs.INPUT.nameroot + ".md5")

baseCommand: [md5sum]
```

ex.yml:

```
INPUT:
  class: File
  location: /home/jeremiah/XX_UUID_XX.db
```

**Comment by hannes-ucsc** [comment]
I don't think we can expect Toil's CWL runner to work with any version of `cwltool`. In fact, Toil's `setup.py` explicitly pins the `cwltool` requirement to 1.0.20160714182449.

What's the reasoning behind upgrading `cwltool` to `1.0_gdc_e`?

**Comment by jeremiahsavage** [comment]
Sorry to confuse the issue. I see this problem even with the pinned version of `cwltool`
```
(toil)ubuntu@frontend001:~$ cwltool 
/home/ubuntu/.virtualenvs/toil/bin/cwltool 1.0.20160714182449

CWL document required, try --help for details
(toil)ubuntu@frontend001:~$ cwltoil ~/cocleaning-cwl/tools/md5sum.cwl ex.yml 
frontend001 2016-10-07 23:43:49,327 MainThread INFO toil.lib.bioio: Root logger is at level 'INFO', 'toil' logger at level 'INFO'.
frontend001 2016-10-07 23:43:50,026 MainThread INFO toil.lib.bioio: Root logger is at level 'INFO', 'toil' logger at level 'INFO'.
frontend001 2016-10-07 23:43:50,028 MainThread INFO toil.jobStores.fileJobStore: Path to job store directory is '/tmp/tmpRM2Mdv'.
frontend001 2016-10-07 23:43:50,028 MainThread INFO toil.jobStores.abstractJobStore: The workflow ID is: '965e75e7-8d1b-405a-ac4e-2395b1d75410'
Traceback (most recent call last):
  File "/home/ubuntu/.virtualenvs/toil/bin/cwltoil", line 11, in <module>
    load_entry_point('toil', 'console_scripts', 'cwltoil')()
  File "/home/ubuntu/toil/src/toil/cwl/cwltoil.py", line 653, in main
    builder = t._init_job(job, basedir=basedir) #, use_container=use_container)
  File "/home/ubuntu/.virtualenvs/toil/local/lib/python2.7/site-packages/cwltool/process.py", line 420, in _init_job
    raise WorkflowException("Document has DockerRequirement under 'requirements' but use_container is false.  DockerRequirement must be under 'hints' or use_container must be true.")
cwltool.errors.WorkflowException: Document has DockerRequirement under 'requirements' but use_container is false.  DockerRequirement must be under 'hints' or use_container must be true.
```

I am separately attempting to see if toil can run with other `cwltool` due to issues such as https://github.com/common-workflow-language/cwltool/pull/170 which are not yet fixed in upstream, but are patched in a local branch.

**Comment by jeremiahsavage** [comment]
With the patch https://github.com/BD2KGenomics/toil/pull/1201 :
```
(toil)ubuntu@frontend001:~$ cwltool 
/home/ubuntu/.virtualenvs/toil/bin/cwltool 1.0.20160714182449

CWL document required, try --help for details
(toil)ubuntu@frontend001:~$ cwltoil ~/cocleaning-cwl/tools/md5sum.cwl ex.yml 
frontend001 2016-10-07 23:51:50,914 MainThread INFO toil.lib.bioio: Root logger is at level 'INFO', 'toil' logger at level 'INFO'.
frontend001 2016-10-07 23:51:51,609 MainThread INFO toil.lib.bioio: Root logger is at level 'INFO', 'toil' logger at level 'INFO'.
frontend001 2016-10-07 23:51:51,611 MainThread INFO toil.jobStores.fileJobStore: Path to job store directory is '/tmp/tmpaDQIig'.
frontend001 2016-10-07 23:51:51,611 MainThread INFO toil.jobStores.abstractJobStore: The workflow ID is: '8b80a9bf-be75-4a8b-b03a-095cf71cc3eb'
frontend001 2016-10-07 23:51:51,617 MainThread INFO toil.jobStores.abstractJobStore: Unable to import 'toil.jobStores.azureJobStore'. You may want to try reinstalling Toil with additional extras.
frontend001 2016-10-07 23:51:51,619 MainThread INFO toil.jobStores.abstractJobStore: Unable to import 'toil.jobStores.googleJobStore'. You may want to try reinstalling Toil with additional extras.
frontend001 2016-10-07 23:51:51,621 MainThread INFO toil.jobStores.abstractJobStore: Unable to import 'toil.jobStores.aws.jobStore'. You may want to try reinstalling Toil with additional extras.
frontend001 2016-10-07 23:51:51,640 MainThread INFO toil.common: Using the single machine batch system
frontend001 2016-10-07 23:51:51,640 MainThread WARNING toil.batchSystems.singleMachine: Limiting maxCores to CPU count of system (8).
frontend001 2016-10-07 23:51:51,640 MainThread WARNING toil.batchSystems.singleMachine: Limiting maxMemory to physically available memory (50651033600).
frontend001 2016-10-07 23:51:51,640 MainThread INFO toil.common: Created the workflow directory at /tmp/toil-8b80a9bf-be75-4a8b-b03a-095cf71cc3eb
frontend001 2016-10-07 23:51:51,640 MainThread WARNING toil.batchSystems.singleMachine: Limiting maxDisk to physically available disk (3327827968).
frontend001 2016-10-07 23:51:51,641 MainThread INFO toil.batchSystems.singleMachine: Setting up the thread pool with 80 workers, given a minimum CPU fraction of 0.100000 and a maximum CPU value of 8.
frontend001 2016-10-07 23:51:51,661 MainThread INFO toil.common: Written the environment for the jobs to the environment file
frontend001 2016-10-07 23:51:51,661 MainThread INFO toil.common: Caching all jobs in job store
frontend001 2016-10-07 23:51:51,661 MainThread INFO toil.common: 0 jobs downloaded.
frontend001 2016-10-07 23:51:51,664 MainThread INFO toil.realtimeLogger: Real-time logging disabled
frontend001 2016-10-07 23:51:51,667 MainThread INFO toil.leader: (Re)building internal scheduler state
frontend001 2016-10-07 23:51:51,667 MainThread INFO toil.leader: Checked batch system has no running jobs and no updated jobs
frontend001 2016-10-07 23:51:51,667 MainThread INFO toil.leader: Found 1 jobs to start and 0 jobs with successors to run
frontend001 2016-10-07 23:51:51,668 MainThread INFO toil.leader: Starting the main loop
frontend001 2016-10-07 23:51:51,668 Thread-1 INFO toil.batchSystems.singleMachine: Executing command: '/home/ubuntu/.virtualenvs/toil/bin/_toil_worker file:/tmp/tmpaDQIig T/d/jobh2g2hn'.
frontend001 2016-10-07 23:52:02,939 MainThread INFO toil.leader: No jobs left to run so exiting.
frontend001 2016-10-07 23:52:02,940 MainThread INFO toil.leader: Finished the main loop
frontend001 2016-10-07 23:52:02,940 MainThread INFO toil.leader: Waiting for stats and logging collator thread to finish ...
frontend001 2016-10-07 23:52:03,190 Thread-82 INFO toil.leader: Got message from job at time 10-07-2016 23:52:03: Successfully deleted local copies of file with ID 'R/Z/tmpL_83YW.tmp'.
frontend001 2016-10-07 23:52:03,190 Thread-82 INFO toil.leader: Got message from job at time 10-07-2016 23:52:03: Successfully deleted local copies of file with ID 'A/W/tmpzM9mU1.tmp'.
frontend001 2016-10-07 23:52:03,191 MainThread INFO toil.leader: ... finished collating stats and logs. Took 0.25093793869 seconds
frontend001 2016-10-07 23:52:03,192 MainThread INFO toil.leader: Waiting for service manager thread to finish ...
frontend001 2016-10-07 23:52:03,671 MainThread INFO toil.leader: ... finished shutting down the service manager. Took 0.479404211044 seconds
frontend001 2016-10-07 23:52:03,672 MainThread INFO toil.leader: Finished toil run successfully
{
    "OUTPUT": {
        "checksum": "sha1$31be2d1a5a62e193e2404510996f896a0309196b", 
        "basename": "picard-tools-2.5.0.md5", 
        "location": "file:///tmp/tmpwdXJo9/picard-tools-2.5.0.md5", 
        "path": "/home/ubuntu/picard-tools-2.5.0_2_3_4.md5", 
        "class": "File", 
        "size": 110
    }
frontend001 2016-10-07 23:52:03,683 MainThread INFO toil.common: Attempting to delete the job store
frontend001 2016-10-07 23:52:03,686 MainThread INFO toil.common: Successfully deleted the job store
```
